src/GitVersionCore/GitPreparer.cs

```diff
@@ -7,66 +7,100 @@
 
 namespace GitVersion
 {
-    public class GitPreparer
+    public class GitPreparer : IGitPreparer
     {
         private readonly ILog log;
-        private string dynamicRepositoryLocation;
-        private AuthenticationInfo authentication;
-        private bool noFetch;
+        private readonly string dynamicRepositoryLocation;
+        private readonly AuthenticationInfo authentication;
+        private readonly bool noFetch;
 
         private const string defaultRemoteName = "origin";
+        private string dynamicGitRepositoryPath;
 
         public GitPreparer(ILog log, Arguments arguments)
-            : this(log, arguments.TargetUrl, arguments.DynamicRepositoryLocation, arguments.Authentication, arguments.NoFetch, arguments.TargetPath)
-        {
-        }
-
-        public GitPreparer(ILog log, string targetUrl, string dynamicRepositoryLocation, Authentication authentication, bool noFetch, string targetPath)
         {
             this.log = log;
-            TargetUrl = targetUrl;
-            this.dynamicRepositoryLocation = dynamicRepositoryLocation;
-            this.authentication =
-                new AuthenticationInfo
-                {
-                    Username = authentication?.Username,
-                    Password = authentication?.Password
-                };
-            this.noFetch = noFetch;
-            WorkingDirectory = targetPath.TrimEnd('/', '\\');
-        }
 
-        public string TargetUrl { get; }
+            TargetUrl = arguments.TargetUrl;
+            WorkingDirectory = arguments.TargetPath.TrimEnd('/', '\\');
 
-        public string WorkingDirectory { get; }
-
-        private bool IsDynamicGitRepository => !string.IsNullOrWhiteSpace(DynamicGitRepositoryPath);
+            dynamicRepositoryLocation = arguments.DynamicRepositoryLocation;
+            authentication = new AuthenticationInfo
+            {
+                Username = arguments.Authentication?.Username,
+                Password = arguments.Authentication?.Password
+            };
 
-        private string DynamicGitRepositoryPath { get; set; }
+            noFetch = arguments.NoFetch;
+        }
 
-        public void Initialise(bool normalizeGitDirectory, string currentBranch, bool shouldCleanUpRemotes = false)
+        public void Initialize(bool normalizeGitDirectory, string currentBranch, bool shouldCleanUpRemotes = false)
```

Comment:
Is perhaps `Prepare()` a better method name for the main method of a `GitPreparer` class?

Reply:
Changed to `Prepare`